docs/overview/release-lifecycle.md

```diff
@@ -0,0 +1,222 @@
+  of Zulip][fork-zulip].
+
+### Compatibility and upgrading
+
+A Zulip design goal is for there never to be a reason to run an old
+version of Zulip. We work extremely hard to make sure Zulip is stable
+for self-hosters, has no regressions, and that the [Zulip upgrade
+process](../production/upgrade-or-modify.md) Just Works.
+
+The Zulip server and clients apps are all carefully engineered to
+ensure compatibility with old versions.  In particular:
+
+* The Zulip mobile and desktop apps maintain backwards-compatibility
+  code to support any Zulip server since 2.1.0.  (They may also work
+  with older versions, with a degraded experience).
+* Zulip maintains an [API changelog](https://zulip.com/api/changelog)
+  detailing all changes to the API to make it easy for client
+  developers to do this correctly.
+* The Zulip server preserves backwards-compatibility in its API to
+  support versions of the mobile and desktop apps released in roughly
+  the last year. Because these clients auto-update, generally there
+  are only a handful of active clients left by the time we desupport a
+  version.
+
+As a result, we generally do not backport changes to previous stable
+release series except in rare cases involving a security issue or
+critical bug just after publishing a major release.
+
+[blog-major-releases]: https://blog.zulip.com/tag/major-releases/
+[upgrade-from-git]: ../production/upgrade-or-modify.html#upgrading-from-a-git-repository
+
+### Security releases
+
+When we discover a security issue in Zulip, we publish a security and
+bug fix release, transparently documenting the issue(s) using the
+industry-standard [CVE advisory process](https://cve.mitre.org/).
+
+When new security releases are published, we simultaneously publish
+the fixes to the `master` and stable release branches (E.g. `4.x`), so
+that anyone using those branches can immediately upgrade as well.
+
+See also our [security model][security-model] documentation.
+
+[security-model]: ../production/security-model.md
+
+### Upgrade nag
+
+Starting with Zulip 4.0, the Zulip webapp will display a banner
+warning users of a server running a Zulip release that is more than 18
+months old. We do this for a few reasons:
+
+* It is unlikely that a server of that age is not vulnerable to
+  a security bug in Zulip or one of its dependencies.
+* The Zulip mobile and desktop apps are only guaranteed to support
+  server versions less than 18 months old.
+
+The nag will appear only to organization administrators starting a
+month before the deadline; after that, it will appear for all users on
+the server.
+
+You can adjust the deadline for your installation by setting e.g.
+`SERVER_UPGRADE_NAG_DEADLINE_DAYS = 30 * 21` in
+`/etc/zulip/settings.py` and then [restarting the server](../production/settings.md).
+
+### Operating system support
+
+Zulip supports all currently supported versions of the upstream
+operating systems one can install it on.  We document how to correctly
```

Comment:
I think I know what the above sentence means, but it's very confusing as written.